src/node/desktop/src/main/toolbar-manager.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * toolbar-manager.ts
+ *
+ * Copyright (C) 2022 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+/* eslint-disable max-len */
+
+import { BrowserWindow } from 'electron';
+import * as fs from 'fs';
+
+export interface ToolbarData {
+  buttons: {
+    iconPath: string;
+    tooltip: string;
+    onClick: string;
+  }[];
+}
+
+export class ToolbarManager {
+  constructor() {}
+
+  async createToolbar(window: BrowserWindow, toolbarData: ToolbarData) {
+    const jsScript =
+      'console.log("windowOnload");' +
+      this.addToolbarJsAsText() +
+      ';' +
+      this.addStylesJsAsText() +
+      ';' +
+      this.addButtonsJsAsText(toolbarData);
+
+    try {
+      await window.webContents.executeJavaScript(jsScript);
+
+      // eslint-disable-next-line @typescript-eslint/no-implicit-any-catch
+    } catch (err: any) {
+      if (err.message !== 'An object could not be cloned.') {
+        console.log('Error message: ', err.message);
+        console.error('Error ', err, ' when trying to create toolbar with data: ', toolbarData);
+        console.log('---------------------------');
+        console.log('JS Script for Toolbar: ');
+        console.log(jsScript);
+      }
+    }
+  }
+
+  addToolbarJsAsText() {
+    const toolbar = ` 
+      const toolbar = document.createElement('div');
```

Comment:
Toolbar needs to be annotated as such for accessibility.

At a minimum the outer div needs `role="toolbar"` and `aria-label="Navigation"`.

Officially a toolbar should support moving between buttons with arrow keys, but we don't support that elsewhere yet, so as long as you can move between buttons with the tab keys that is fine.

Reply:
https://www.w3.org/TR/wai-aria-practices/#toolbar

Reply:
Done in https://github.com/rstudio/rstudio/pull/10541/commits/6deede4405d4f6cfc08f5c0d2759f7ae4a7d7c6e